src/main/java/ParserStorageUi/Storage.java

```diff
@@ -0,0 +1,107 @@
+package parserstorageui;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import exceptions.DukeException;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+public class Storage {
+
+    /**
+     * The file path of the folder of duke.txt
+     **/
+    private final String filePath;
+
+    /**
+     * The object file path for duke.txt
+     **/
+    private Path file;
+
+    /**
+     * Initializes Storage with filePath as the parameter
+     *
+     * @param filePath
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        initiateFile();
+    }
+
+    /**
+     * Initiate the file
+     **/
+    private void initiateFile() {
+        try {
+            if (Files.notExists(Paths.get(filePath))) {
+                Files.createDirectory(Paths.get(filePath));
+            }
+            if (Files.notExists(Paths.get(filePath + "/duke.txt"))) {
+                file = Files.createFile(Paths.get(filePath + "/duke.txt"));
+            } else {
+                file = Paths.get(filePath + "/duke.txt");
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Load all the tasks from the save file to ArrayList<Task>
+     **/
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> temp = new ArrayList<>();
```

Comment:
I think according to coding standards they recommended that we put plural form for collections such as lists, so here maybe 'tasks' would be a better variable name!